### PR TITLE
guess who? by Aschwin

### PR DESCRIPTION
**Pull request by AschwinSiaila**
https://suspicious-hodgkin-de6ef3.netlify.app/

I was happy to see my audio was doing great but i turns out that my audio works on my server but not on netlify. I did not work that out. I liked this project and i would like to change the characters in this game.

### CODE REVIEW

File: code/script.js [hunk 1]
```diff
@@ -218,108 +228,150 @@ const generateBoard = () => {
...
+  const { category, value } = currentQuestion;
+  // Show the correct alert message for different categories
+  if (keep) {
+    alert(
+      `Yes, the person has ${value} in the category ${category}! Keep all the people with ${value} in ${category}`
+    );
+  } else {
+    alert(
+      `No, the person doesn't have ${value} in the category ${category}! Remove all people with ${value} in category ${category}`
+    );
```
Comment (nehrwein) [inline code/script.js:292]: supertidy and smart solution for both checkQuestion() and alertOfUser()

File: code/script.js [hunk 2]
```diff
@@ -218,108 +228,150 @@ const generateBoard = () => {
...
   // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
+  if (secret.name === personToCheck) {
+    // 2. Set a Message to show in the win or lose section accordingly
+    winOrLoseSection.style.display = "block";
+    board.style.display = "none";
+    winAudio.play();
```
Comment (nehrwein) [inline code/script.js:347]: This is, what I've done for the audio. Maybe it helps :)

```
if (personToCheck === secret.name) {
    //applause if winning
    let audio = document.createElement("audio");
    audio.src = "./assets/applause.wav";
    audio.play();
```

File: code/script.js [hunk 3]
```diff
@@ -218,108 +228,150 @@ const generateBoard = () => {
...
+      "Sorry 🙁 inspector this is not the secret person, but don't give up and try again";
+  }
+};
+// 3. Show the win or lose section
+
+const playAgain = () => {
+  winOrLoseSection.style.display = "none";
+  board.style.display = "flex";
+
+  location.reload();
```
Comment (nehrwein) [inline code/script.js:364]: What does this `location.reload();` do?

**Review by nehrwein — COMMENTED** [review]
Hej Aschwin, 👋 
I really enjoyed doing this review on your code. I have to say, that I am impressed of how well you use the js-logic by implementing smart functions and thereby producing less lines of code. 
Chapeau! 🎩

You did all the blue steps and even added some red and black ones. I made a suggestion for an audio fix in my comments, so maybe this can help. The only thing, I am missing is a Read.me file with some background information on your project.

Overall a really great work! Be proud!
//Birgit

File: code/index.html [hunk 4]
```diff
@@ -20,24 +20,41 @@
         alt="Guess Who"
       />
 
-      <h1>Does the person have</h1>
+      <h1>Inspector, can you find out?</h1>
 
       <select id="questions">
         <optgroup label="hair">
+          <option value="">Guess here! 🕵️‍♂️👆🕵️‍♀️</option>
```
[the remark this answers is not in the document]
Comment (Rephili) [reply code/index.html:27]: I agree with Birgit!

File: code/index.html [hunk 5]
```diff
@@ -20,24 +20,41 @@
...
         </optgroup>
         <optgroup label="accessories">
           <option value="glasses">glasses</option>
+          <option value="hat">hat</option>
+          <option value="sunglasses">sunglasses</option>
           <!-- Add the other accessory option here. (hat)  -->
         </optgroup>
         <optgroup label="other">
           <option value="smoker">a smoking habit</option>
+          <option value="beard">beard</option>
```
Comment (Rephili) [inline code/index.html:55]: Great job, didn't occur to me to add beard and male/female. :)

File: code/script.js [hunk 6]
```diff
@@ -1,213 +1,223 @@
...
 
 // Global variables
-let secret
-let currentQuestion
-let charactersInPlay
+let secret;
+let currentQuestion;
+let charactersInPlay;
+let winAudio = new Audio(
+  "http://soundfxcenter.com/music/television-theme-songs/8d82b5_Inspector_Gadget_Ending_Theme_Song.mp3"
```
Comment (Rephili) [inline code/script.js:212]: Nice extra effect with different audios depending on the outcome of the game

File: code/script.js [hunk 7]
```diff
@@ -218,108 +228,150 @@ const generateBoard = () => {
...
-  const { category, value } = currentQuestion
-
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
+  const { category, value } = currentQuestion;
+  console.log(currentQuestion);
```
Comment (Rephili) [inline code/script.js:268]: the console.logs could be removed :)

**Review by Rephili — COMMENTED** [review]
Hi Aschwin! I didn't have any project to make a code review on this week so I decided to have a look at yours. :) I can see that Birgit already has written some great comments and I agree with her on how structured you are with your code, it is tidy and easy to read. I have tried out your game and everything seems to be working as it should. Awesome that you had time to add the sound effects and some extra values to the characters. Good job!

**Pull request closed without merging**